test/extensions/filters/http/common/fuzz/uber_per_filter.cc

```diff
@@ -176,6 +177,10 @@ void UberFilterFuzzer::perFilterSetup() {
   ON_CALL(decoder_callbacks_, decodingBuffer()).WillByDefault([this]() -> const Buffer::Instance* {
     return decoding_buffer_;
   });
+
+  // For filters that need main event loop.
+  ON_CALL(encoder_callbacks_, dispatcher()).WillByDefault(testing::ReturnRef(*main_dispatcher_));
+  ON_CALL(decoder_callbacks_, dispatcher()).WillByDefault(testing::ReturnRef(*main_dispatcher_));
```

Comment:
I think this will prevent filters that use the dispatcher and don't have `with_main_event_loop_` set to true from executing their dispatcher callback code.

I'm contemplating whether the file system buffer filter will (option 1) benefit from its own fuzzer (like we do for ext_authz and ext_proc for example), or (option 2) whether we can hack this by setting these ON_CALLs in the `UberFilterFuzzer::fuzz` method if this filter is being tested, and resetting it when the method ends.

For now let's take option 2 if possible.

Reply:
The `ON_CALL`s are now only inserted, when the flag is set.